x-pack/test/detection_engine_api_integration/security_and_spaces/tests/create_rules.ts

```diff
@@ -96,7 +97,24 @@ export default ({ getService }: FtrProviderContext) => {
           .expect(200);
 
         // wait for Task Manager to execute the rule and update status
-        await new Promise((resolve) => setTimeout(resolve, 5000));
+        await waitFor(async () => {
+          const { body: statusBody } = await supertest
+            .post(DETECTION_ENGINE_RULES_STATUS_URL)
+            .set('kbn-xsrf', 'true')
+            .send({ ids: [body.id] })
+            .expect(200);
+          return statusBody[body.id].current_status?.status === 'succeeded';
+        });
+
+        await waitFor(async () => {
+          const { body: statusBody } = await supertest
+            .post(DETECTION_ENGINE_RULES_STATUS_URL)
+            .set('kbn-xsrf', 'true')
+            .send({ ids: [body.id] })
+            .expect(200);
+          return statusBody[body.id].current_status?.status === 'succeeded';
+        });
+
```

Comment:
Am I reading this right, that there's now three separate calls to `DETECTION_ENGINE_RULES_STATUS_URL`, two of which are wrapped in `waitFor`? Should there just be one call to `DETECTION_ENGINE_RULES_STATUS_URL` wrapped in a `waitFor`? Perhaps a little find/replace magic here? 🍝

Reply:
Ohh, I think I see now based off the other examples. There should be one `expect` wrapped in a `waitFor` for the _polling_, and then a subsequent one for the final verification?

Reply:
Correct.